bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gems/CanopusWfs.java

```diff
@@ -0,0 +1,191 @@
+    }
+
+    @Override
+    public GuideStarValidation validate(final SPTarget guideStar, final ObsContext ctx) {
+        final Option<Long> when = ctx.getSchedulingBlockStart();
+        return guideStar.getSkycalcCoordinates(when).map(coords ->
+                areProbesInRange(coords, ctx) ? GuideStarValidation.VALID : GuideStarValidation.INVALID
+        ).getOrElse(GuideStarValidation.UNDEFINED);
+    }
+
+    /**
+     * Returns an Area representing the probe range. This is used in drawing the probe range.
+     */
+    public static Area probeRange(final ObsContext ctx) {
+        return offsetIntersection(ctx, ctx.getSciencePositions());
+    }
+
+    /**
+     * Check if the primary guide star is in range from the given offset
+     */
+    @Override
+    public boolean inRange(final ObsContext ctx, final Offset offset) {
+        return ctx.getTargets().getPrimaryGuideProbeTargets(this)
+                .flatMap(GuideProbeTargets::getPrimary)
+                .flatMap(gs -> gs.getSkycalcCoordinates(ctx.getSchedulingBlockStart())
+                        .map(gscoords -> areProbesInRangeWithOffsets(gscoords, ctx, Collections.singleton(offset)))
+                ).getOrElse(false);
+    }
+
+    /**
+     * Determines if the guide probe can reach the provided coordinates in the given observing context (if any).
+     */
+    public static boolean areProbesInRange(final Coordinates coords, final ObsContext ctx) {
+        return areProbesInRangeWithOffsets(coords, ctx, ctx.getSciencePositions());
+    }
+
+    /**
+     * Helper method to extract the common code from inRange and areProbesInRange.
+     */
+    private static boolean areProbesInRangeWithOffsets(final Coordinates coords, final ObsContext ctx, final Set<Offset> offsets) {
+        return ctx.getBaseCoordinates().map(bcs -> {
+            // Calculate the difference between the coordinate and the observation's base position.
+            final CoordinateDiff diff = new CoordinateDiff(bcs, coords);
+
+            // Get offset and switch it to be defined in the same coordinate system as the shape.
+            final Offset dis = diff.getOffset();
+            final double p = -dis.p().toArcsecs().getMagnitude();
+            final double q = -dis.q().toArcsecs().getMagnitude();
+
+            return offsetIntersection(ctx, offsets).contains(p, q);
+        }).getOrElse(false);
+    }
+
+    /**
+     * Gets the intersection of the FOV and the specified offsets.
+     */
+    private static Area offsetIntersection(final ObsContext ctx, final Set<Offset> offsets) {
+        final double t = ctx.getPositionAngle().toRadians();
+
+
+        return offsets.stream().map(pos -> {
+            final double p = pos.p().toArcsecs().getMagnitude();
+            final double q = pos.q().toArcsecs().getMagnitude();
+
+            final AffineTransform xform = new AffineTransform();
+            if (t != 0.0) xform.rotate(-t);
```

Comment:
This could be slightly clearer if we use directly the position angle like
`if (pa != PositionAngle.zero) xform.rotate(-pa.toRadians)`

Reply:
I agree. I'll make that change.

Reply:
Kind of annoying, because now we are using both the `edu.gemini.spModel.core.Angle` and the `edu.gemini.skycalc.Angle`, and this is Java, so I will have to preface the former with the fully qualified package name.